cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/TetherStore.java

```diff
@@ -0,0 +1,215 @@
+      fields.add(Fields.stringField(StoreDefinition.TetherStore.PEER_NAME_FIELD, peerName));
+      fields.add(Fields.stringField(StoreDefinition.TetherStore.TETHER_STATE_FIELD, tetherStatus.toString()));
+      fields.add(Fields.longField(StoreDefinition.TetherStore.LAST_CONNECTION_TIME_FIELD, System.currentTimeMillis()));
+      StructuredTable tetherTable = context.getTable(StoreDefinition.TetherStore.TETHER);
+      tetherTable.update(fields);
+    });
+  }
+
+  /**
+   * Updates tether status for a peer.
+   *
+   * @param peerName name of the peer
+   * @param tetherStatus status of tether with the peer
+   * @throws IOException if updating the table fails
+   */
+  public void updatePeerStatus(String peerName, TetherStatus tetherStatus) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      Collection<Field<?>> fields = new ArrayList<>();
+      fields.add(Fields.stringField(StoreDefinition.TetherStore.PEER_NAME_FIELD, peerName));
+      fields.add(Fields.stringField(StoreDefinition.TetherStore.TETHER_STATE_FIELD, tetherStatus.toString()));
+      StructuredTable tetherTable = context.getTable(StoreDefinition.TetherStore.TETHER);
+      tetherTable.update(fields);
+    });
+  }
+
+  /**
+   * Updates the last connection timestamp for the peer.
+   *
+   * @param peerName name of the peer
+   * @throws IOException if updating the table fails
+   */
+  public void updatePeerTimestamp(String peerName) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      Collection<Field<?>> fields = new ArrayList<>();
+      fields.add(Fields.stringField(StoreDefinition.TetherStore.PEER_NAME_FIELD, peerName));
+      fields.add(Fields.longField(StoreDefinition.TetherStore.LAST_CONNECTION_TIME_FIELD, System.currentTimeMillis()));
+      StructuredTable tetherTable = context.getTable(StoreDefinition.TetherStore.TETHER);
+      tetherTable.update(fields);
+    });
+  }
+
+  /**
+   * Deletes a peer
+   *
+   * @param peerName name of the peer
+   * @throws IOException if deleting the table fails
+   */
+  public void deletePeer(String peerName) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.TetherStore.TETHER);
+      capabilityTable
+        .delete(Collections.singleton(Fields.stringField(StoreDefinition.TetherStore.PEER_NAME_FIELD, peerName)));
+    });
+  }
+
+  /**
+   * Get information about all tethered peers
+   *
+   * @return information about status of tethered peer
+   * @throws IOException if reading from the database fails
+   */
+  public List<PeerInfo> getPeers() throws IOException {
+    List<PeerInfo> peers = new ArrayList<>();
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable tetherTable = context
+        .getTable(StoreDefinition.TetherStore.TETHER);
+      CloseableIterator<StructuredRow> iterator = tetherTable
```

Comment:
You need to close the iterator after use:

```
try (CloseableIterator<StructuredRow> iter = ...) {
   // Use 
}

Reply:
thanks, fixed